### ISSUE TITLE
Silently fails to run AoT version of code when using the promise returned from bootstrapModule

### ISSUE THREAD

**Issue by chasemgray** [UltimateAngular/aot-loader#6]
We have our bootstrapModule inside of a function and we then call the .then function of the returned promise to set a few variables. This seems to silently fail when using the aot-loader. If I move bootstrapModule outside it seems to work.

```
export function main() {
  platformBrowserDynamic().bootstrapModule(AppModule)
    .then((upgradeAdapterRef) => {
      registerInjector(upgradeAdapterRef.injector);
      window.ngApp = upgradeAdapterRef;
    });
}

bootloader(main);
```

**Comment by rynclark** [comment]
Hey, have you got an example repo I can take a look at? Cheers

**Comment by rynclark** [comment]
I think I know what's going down already but it would be great if you do.

**Comment by chasemgray** [comment]
Is there a sample somewhere with aot-loader already set up I can modify as an example or do I need to get a sample app up and running from scratch?

**Comment by rynclark** [comment]
I'll get it replicated and fixed tomorrow morning 😊

**Comment by chasemgray** [comment]
k, in the meantime I can just not set those variables while I get everything working.

**Comment by chasemgray** [comment]
works if I switch it to
```
  const bootstrapPromise = platformBrowserDynamic().bootstrapModule(AppModule);
  bootstrapPromise.then((upgradeAdapterRef) => {
      registerInjector(upgradeAdapterRef.injector);
      window.ngApp = upgradeAdapterRef;
    });
```

**Comment by rynclark** [comment]
Just fixed in 0.1.9 - the AST wasn't searching recursively for `platformBrowserDynamic` calls so it wasn't picking them up inside your `main` function.

**Comment by chasemgray** [comment]
thanks!
